sdk/core/core-rest-pipeline/test/defaultLogPolicy.spec.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { DEFAULT_RETRY_POLICY_COUNT } from "../src/constants";
+import { PipelinePolicy } from "../src/pipeline";
+import { assert } from "chai";
+import { createHttpHeaders } from "../src/httpHeaders";
+import { createPipelineFromOptions } from "../src/createPipelineFromOptions";
+import { createPipelineRequest } from "../src/pipelineRequest";
+import { isNode } from "@azure/core-util";
+import sinon from "sinon";
+
+describe("defaultLogPolicy", function () {
+  it("should be invoked on every retry", async function () {
+    const request = createPipelineRequest({
+      url: "https://bing.com",
+    });
+
+    const testSignPolicy: PipelinePolicy = {
+      name: "testSignPolicy",
+      sendRequest: async function (req, next) {
+        const response = await next(req);
+        return response;
+      },
+    };
+
+    const pipeline = createPipelineFromOptions({
+      retryOptions: { maxRetryDelayInMs: 0 },
+    });
+    pipeline.addPolicy(testSignPolicy, { phase: "Sign" });
+
+    const orderedPolicies = pipeline.getOrderedPolicies();
+
+    const expectedOrderedPolicies = isNode ? ["proxyPolicy", "decompressResponsePolicy"] : [];
+    expectedOrderedPolicies.push(
+      "formDataPolicy",
+      "userAgentPolicy",
+      "setClientRequestIdPolicy",
+      "defaultRetryPolicy",
+      "tracingPolicy"
+    );
+    if (isNode) {
+      expectedOrderedPolicies.push("redirectPolicy");
+    }
+    expectedOrderedPolicies.push("testSignPolicy", "logPolicy");
+    assert.deepEqual(
+      orderedPolicies.map((policy) => policy.name),
+      expectedOrderedPolicies
+    );
+
+    const order: string[] = [];
+    orderedPolicies.map((policy) => {
```

Comment:
nit: it doesn't look like you are using the return array from `orderedPolicies.map` when you don't need the resulting array you have 2 options `orderedPolicies.forEach` or 

```ts
for (const policy of orderedPolicies){
   //...
}
```

I personally like the for of better, but no strong opinions

Reply:
my personal preference is also for/of, avoiding the unnecessary closure is nice especially since let/const have block scoping now anyway

Reply:
Done!